Blogpost about saving network traffic (#314)
deltachat/deltachat-pages: _posts/2020-03-10-reduced-message-size.md

---

> title: Saving Network Traffic - Delta's Efforts to Reduce the Message Size

Reviewer: Reduced Network Traffic and Message Size

> title: Saving Network Traffic - Delta's Efforts to Reduce the Message Size

Reviewer: ```suggestion
title:  Reduced Network Traffic and Message Size
```

Author: Why do you think this would be a better headline? In my opinion, the purpose of a headline is to make people click on the link; yes, technically network traffic was also reduced directly, and not only by reducing the message size... but a headline doesn't need to be overly descriptive, it needs to tell a story.

Reviewer 2: There is an ingress in the headline, and a dash in place of an em-dash.
Nothing is clickable in the original, it is just a dead giveaway. It does the opposite, you don't have to click if you know the whole story.

author: compl4xx
image: ../assets/blog/screenshots/2020-03-10-delta-chat-message-size-wifi-charger.jpg
---

**In the last months, the Delta Chat developers put a lot of focus on saving

> traffic. The result: using Delta Chat consumes far less data now.**

Reviewer: ```suggestion
**In the last months, the Delta Chat developers put a lot of focus on saving
traffic. The result: current Delta Chat mobile releases consumes far less data now.**
```

Author: good point :)

It's easy to forget that many people still pay data plans by the GB - their
data traffic is **expensive, limited, and slow.** For many people, it's even a
question of money; Internet traffic can be quite expensive. In such moments,
this can be the critical difference: is an app usable at all, or does it use
too much traffic?

In the effort to save network traffic, the developers improved Delta Chat a lot
in the last months. This helps everyone with an expensive data plan. It's also
useful if your Internet connection is throttled because you ran out of data,

> but the month isn't over yet. 

Reviewer: this paragraph can be removed without loosing information that isn't already in the previous paragraph.

Author: Agreed.

![A "free Wi-Fi inside" sign at a restaurant.](../assets/blog/screenshots/2020-03-10-delta-chat-message-size-wifi-charger.jpg)

> *Somehow free Wi-Fi became one of the main offerings of Cafés and Restaurants.*

Reviewer: i still think it's a too verbose intro. 

Also i don't get the relevance of the free-wifi image for reducing network traffic.  If WiFi is free why would i care for reduced network traffic?

Author: Well, people with low data are always looking for a free Wi-Fi spot... I think it's very obvious. This could mean that they don't have to find one anymore, at least for Delta. Especially if you have no Wi-Fi at home, but only a data plan...

Author: I changed the image description to make the connection more obvious.

taking minutes to arrive. For many people, it's even a question of money;
Internet traffic can be quite expensive. But does it have to be this way?

In the effort to save network traffic, the developers improved Delta Chat a lot
in the last months. This helps everyone with an expensive data plan. It's
also useful if your Internet connection is throttled because you ran out of
data, but the month isn't over yet. In regions with bad Internet connection
it's always good not to burden it too much.

![A "free Wi-Fi inside" sign at a restaurant.](../assets/blog/screenshots/2020-03-10-delta-chat-message-size-wifi-charger.jpg)

*Somehow free Wi-Fi became one of the main offerings of Cafés and Restaurants.*

It's easy to forget that many people still pay data plans by the GB - their
data traffic is **expensive, limited, and slow.** They all benefit from a
smaller message size. 

But everyone knows situations with low bandwidth, like on a plane Wi-Fi, on a
train, or in places where many people are online. In such moments, this can be
the critical difference: is an app usable at all, or does it use too much
traffic?

## Which Improvements are Saving the Most Traffic?

> 

Reviewer: I don't think we need to make a very big multi-paragraph argument for why data-saving is useful. So I think this whole section starting after the bold text could be substituted with something like this para:  

"While it's typically not a concern for resource-rich users reducing network traffic means that the app becomes more affordable and usable for less-resourced users.  Moreover, if you are using a volume-limited data plan then you will also appreciate recent improvements to Delta Chat's network traffic data usage. Here are a few key things which the recent Android/iOS releases introduced to reduce traffic, in some cases substantially:"

Author: Okay, I cut it down to 2 paragraphs.

Reviewer: hum, so what has "free wifi" got to do with traffic saving?

A few things which were changed to reduce our data traffic consumption:

- Delta now avoids downloading messages that it doesn't display, e.g. messages
  from blocked contacts. This means:
  - No unnecessary traffic
  - No slowdown when getting online with a busy inbox
- What is more important than blocked contacts is ordinary emails with huge
  attachments. If you use Delta with a normal email address and send PDFs

>   around, Delta used to download them and throw away.

Reviewer: Please shift this most major item to the top and make the "blocked contact's messages are not downloaded at all, irrespective if they are a chat message or not." a little addentum item.

- We optimized the traffic on idle, when the app isn't used. In 10 minutes, it
  used to consume 6 kB - [this was reduced down to 107 bytes.](https://github.com/deltachat/deltachat-core-rust/issues/506)

> - Attachment download is now postponed until you have Wi-Fi, or download it

Reviewer 3: oops, I expected it to go into "What else is possible?" section. It is not implemented in the current release.

Author: ah, good to know ;)

  moment. Before, there was an email for each read receipt.

### Changing the PGP keys to ed25519 Elliptic Curve Keys

We changed newly generated PGP Keys to the new ed25519 algorithm - those keys
are a lot smaller. This is very important, as the keys are sent in every

> message at the moment.

Reviewer: ```suggestion
message at the moment. Note that the current Android/iOS releases still generate RSA2048 keys by default because the Delta Chat testing discovered an incompatibility which was fixed in the current releases.  After the next Desktop release we should finally be able to switch to the much shorter ED25519 keys by default. 
```

![two editor windows; on the left you see an RSA key, on the right an ed25519
key.](../assets/blog/screenshots/2020-03-10-delta-chat-ed25519-key-size-comparison.png)

*On the left, my old RSA key; it doesn't even fit on the screen. On the right,
a new ed25519 key.*

We wanted to go for this way earlier, but we had to avoid compability issues.
Maybe you noticed that we didn't update F-Droid versions for about six months;
with all the old Delta versions around, we couldn't introduce changes like
this without risking compability.

## What Else is Possible? A Glance Into The Future

This is already a start, and saves many people a lot of money, who have to live
with 50 MB per month. But there are even more ideas to reduce the traffic

> Delta Chat generates, which might be implemented soon™:

Reviewer: ```suggestion
While the above already provides for substantial reductions a few more data-saving efforts have been discussed:
```

- Disabling encryption for some chats: making it possible not to send Autocrypt
  headers at all. This is highly controversial though.
- Attachment download could be postponed until you have Wi-Fi, or download it
  manually.
- Sending the key fingerprint in a special header, so we don't always have to
  download the whole Autocrypt header.
- Sending small messages through a header, thereby saving IMAP commands in the
  background.
- Not sending a key per gossip if the key was gossipped by someone else in the
  last 48 hours.
- Maybe we can avoid sending the to/cc headers to groups where everyone uses
  Delta Chat. This is a security feature, but also saves traffic.
- And finally there are some ideas about leaving out the Autocrypt header,
  when encryption *provably* works. If it stops working, it could still be

>   recovered by a simple message exchange.

Reviewer: i'd strike all these points in favor of the above suggested link to the issue tracker as they are all quite technical anyway.
